Include/Types/Color.hpp

```diff
@@ -268,171 +268,97 @@ class ColorBase
-			T max = std::numeric_limits<T>::max();
-			return ColorBase<T>{ T( 0.933 * max ), T( 0.509 * max ), T( 0.933 * max ), T( 1.0 * max ) };
-		} else {
-			return ColorBase<T>{ T( 0.933 ), T( 0.509 ), T( 0.933 ), T( 1.0 ) };
-		}
+		return _PREDEFINED_COLOR(0.933, 0.509, 0.933);
 	};
 
 	static constexpr ColorBase<T> GREEN()
 	{
-		if constexpr( std::is_integral_v<T> ) {
-			T max = std::numeric_limits<T>::max();
-			return ColorBase<T>{ T( 0.0 * max ), T( 1.0 * max ), T( 0.0 * max ), T( 1.0 * max ) };
-		} else {
-			return ColorBase<T>{ T( 0.0 ), T( 1.0 ), T( 0.0 ), T( 1.0 ) };
-		}
+		return _PREDEFINED_COLOR(0.0, 1.0, 0.0);
 	};
 
 	static constexpr ColorBase<T> LIME()
 	{
-		if constexpr( std::is_integral_v<T> ) {
-			T max = std::numeric_limits<T>::max();
-			return ColorBase<T>{ T( 0.196 * max ), T( 0.803 * max ), T( 0.196 * max ), T( 1.0 * max ) };
-		} else {
-			return ColorBase<T>{ T( 0.196 ), T( 0.803 ), T( 0.196 ), T( 1.0 ) };
-		}
+		return _PREDEFINED_COLOR(0.196, 0.803, 0.196);
 	};
 
 	static constexpr ColorBase<T> OLIVE()
 	{
-		if constexpr( std::is_integral_v<T> ) {
-			T max = std::numeric_limits<T>::max();
-			return ColorBase<T>{ T( 0.419 * max ), T( 0.556 * max ), T( 0.137 * max ), T( 1.0 * max ) };
-		} else {
-			return ColorBase<T>{ T( 0.419 ), T( 0.556 ), T( 0.137 ), T( 1.0 ) };
-		}
+		return _PREDEFINED_COLOR(0.419, 0.556, 0.137);
 	};
 
 	static constexpr ColorBase<T> BLUE()
 	{
-		if constexpr( std::is_integral_v<T> ) {
-			T max = std::numeric_limits<T>::max();
-			return ColorBase<T>{ T( 0.0 * max ), T( 0.0 * max ), T( 1.0 * max ), T( 1.0 * max ) };
-		} else {
-			return ColorBase<T>{ T( 0.0 ), T( 0.0 ), T( 1.0 ), T( 1.0 ) };
-		}
+		return _PREDEFINED_COLOR(0.0, 0.0, 1.0);
 	};
 
 	static constexpr ColorBase<T> NAVY()
 	{
-		if constexpr( std::is_integral_v<T> ) {
-			T max = std::numeric_limits<T>::max();
-			return ColorBase<T>{ T( 0.0 * max ), T( 0.0 * max ), T( 0.5 * max ), T( 1.0 * max ) };
-		} else {
-			return ColorBase<T>{ T( 0.0 ), T( 0.0 ), T( 0.5 ), T( 1.0 ) };
-		}
+		return _PREDEFINED_COLOR(0.0, 0.0, 0.5);
 	};
 
 	static constexpr ColorBase<T> CYAN()
 	{
-		if constexpr( std::is_integral_v<T> ) {
-			T max = std::numeric_limits<T>::max();
-			return ColorBase<T>{ T( 0.0 * max ), T( 1.0 * max ), T( 1.0 * max ), T( 1.0 * max ) };
-		} else {
-			return ColorBase<T>{ T( 0.0 ), T( 1.0 ), T( 1.0 ), T( 1.0 ) };
-		}
+		return _PREDEFINED_COLOR(0.0, 1.0, 1.0);
 	};
 
 	static constexpr ColorBase<T> TEAL()
 	{
-		if constexpr( std::is_integral_v<T> ) {
-			T max = std::numeric_limits<T>::max();
-			return ColorBase<T>{ T( 0.0 * max ), T( 0.5 * max ), T( 0.5 * max ), T( 1.0 * max ) };
-		} else {
-			return ColorBase<T>{ T( 0.0 ), T( 0.5 ), T( 0.5 ), T( 1.0 ) };
-		}
+		return _PREDEFINED_COLOR(0.0, 0.5, 0.5);
 	};
 
 	static constexpr ColorBase<T> YELLOW()
+	{
+		return _PREDEFINED_COLOR(1.0, 1.0, 0.0);
+	};
+
+private:
+	static constexpr ColorBase<T> _PREDEFINED_COLOR(double r, double g, double b)
```

Comment:
Could you change `_PREDEFINED_COLOR` into `PREDEFINED_COLOR`.

Underscore followed by upper-case letter is apparently always reserved by C++. Single underscrore followed by small letter is allowed as a class member as I understood it, however I'm clearing the code of all underscore starting names to be consistent.